.github/workflows/format.yml

```diff
@@ -0,0 +1,35 @@
+name: format code
+on:
+  pull_request:
+
+jobs:
+  format-code:
+    name: Format Code
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package: [flutter, dart]
+    defaults:
+      run:
+        working-directory: ${{ matrix.package }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart@6a218f2413a3e78e9087f638a238f6b40893203d # pin@v1
+        if: ${{ matrix.package == 'dart' }}
+      - uses: subosito/flutter-action@48cafc24713cca54bbe03cdc3a423187d413aafa # pin@v2.10.0
+        if: ${{ matrix.package == 'flutter' }}
+
+      - name: pub get
+        run: ${{ matrix.package }} pub get
+
+      - name: format
+        run: dart format .
+
+      - name: fix
+        run: dart fix --apply
+
+      # actions/checkout fetches only a single commit in a detached HEAD state. Therefore
+      # we need to pass the current branch, otherwise we can't commit the changes.
+      # GITHUB_HEAD_REF is the name of the head branch. GitHub Actions only sets this for PRs.
+      - name: Commit & Push
+        run: ../scripts/commit-formatted-code.sh $GITHUB_HEAD_REF
```

Comment:
Can this actually be part of https://github.com/getsentry/sentry-dart/blob/main/.github/workflows/analyze.yml ?
Right now this is only executed by the dart and flutter packages, we have many more packages, so this approach would not work for all the packages.